### ISSUE TITLE
整个训练过程中'eval_loss'都是nan，但是training_loss正常，一直在下降

### ISSUE THREAD

**Issue by litsh** [ymcui/Chinese-LLaMA-Alpaca-2#370]
### 提交前必须检查以下项目

- [X] 请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
- [X] 我已阅读[项目文档](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki)和[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/常见问题)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案。
- [X] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[LangChain](https://github.com/hwchase17/langchain)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)等，同时建议到对应的项目中查找解决方案。

### 问题类型

模型训练与精调

### 基础模型

Chinese-Alpaca-2 (7B/13B)

### 操作系统

Linux

### 详细描述问题

运行run_sft.sh，其内容如下：
lr=1e-4
lora_rank=64
lora_alpha=128
lora_trainable="q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj"
modules_to_save="embed_tokens,lm_head"
lora_dropout=0.05

pretrained_model=/remote-home/tsli/Chinese-LLaMA-Alpaca-2/model
chinese_tokenizer_path=/remote-home/tsli/Chinese-LLaMA-Alpaca-2/model
dataset_dir=/remote-home/tsli/Chinese-LLaMA-Alpaca-2/data/train_data
per_device_train_batch_size=1
per_device_eval_batch_size=1
gradient_accumulation_steps=8
max_seq_length=512
output_dir=/remote-home/tsli/Chinese-LLaMA-Alpaca-2/experiments
validation_file=/remote-home/tsli/Chinese-LLaMA-Alpaca-2/data/fin_alphaca_val.json

deepspeed_config_file=ds_zero2_no_offload.json

torchrun --nnodes 1 --nproc_per_node 8 run_clm_sft_with_peft.py \
    --deepspeed ${deepspeed_config_file} \
    --model_name_or_path ${pretrained_model} \
    --tokenizer_name_or_path ${chinese_tokenizer_path} \
    --dataset_dir ${dataset_dir} \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --per_device_eval_batch_size ${per_device_eval_batch_size} \
    --do_train \
    --do_eval \
    --seed 123 \
    --fp16 \
    --num_train_epochs 10 \
    --lr_scheduler_type cosine \
    --learning_rate ${lr} \
    --warmup_ratio 0.03 \
    --weight_decay 0 \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --save_total_limit 5 \
    --evaluation_strategy steps \
    --eval_steps 4 \
    --save_steps 200 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --preprocessing_num_workers 8 \
    --max_seq_length ${max_seq_length} \
    --output_dir ${output_dir} \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --lora_rank ${lora_rank} \
    --lora_alpha ${lora_alpha} \
    --trainable ${lora_trainable} \
    --lora_dropout ${lora_dropout} \
    --modules_to_save ${modules_to_save} \
    --torch_dtype float16 \
    --validation_file ${validation_file} \
    --load_in_kbits 16 \
    --gradient_checkpointing \
    --ddp_find_unused_parameters False

得到的结果是每一次evaluation得到的loss都是nan：
{'eval_loss': nan, 'eval_runtime': 0.9052, 'eval_samples_per_second': 91.693, 'eval_steps_per_second': 12.152, 'epoch': 8.85}  
但是train_loss是正常的：
{'loss': 0.0727, 'learning_rate': 3.5094519932415417e-07, 'epoch': 9.36}                                                  

### 依赖情况（代码类问题务必提供）

bitsandbytes             0.41.0
peft                     0.5.0
sentencepiece            0.1.97
torch                    2.0.1
transformers             4.31.0

### 运行日志或截图

数据格式：{
    "output": "yes",
    "instruction": "A lie is an assertion that is believed to be false, typically used with the purpose of deceiving or misleading someone. You are tasked with deciding whether a user is lying in a conversation between customer service and the user.\n\nConversation:\nNone，直接挂断或中途挂断\n\nIn the given conversation, is the user lying or not?\n\nOPTIONS:\n-yes -no",
    "input": null
  }

训练结果：
[INFO|trainer.py:3081] 2023-10-26 08:33:06,225 >> ***** Running Evaluation *****
[INFO|trainer.py:3083] 2023-10-26 08:33:06,225 >>   Num examples = 83
[INFO|trainer.py:3086] 2023-10-26 08:33:06,225 >>   Batch size = 1
{'eval_loss': nan, 'eval_runtime': 0.9023, 'eval_samples_per_second': 91.986, 'eval_steps_per_second': 12.191, 'epoch': 9.19}                                                                                                                       
{'loss': 0.0727, 'learning_rate': 3.5094519932415417e-07, 'epoch': 9.36}                                                  
100%|███████████████████████████████████████████████████████████████████████████████████| 110/110 [17:56<00:00,  9.86s/it][INFO|trainer.py:1934] 2023-10-26 08:33:26,277 >> 

Training completed. Do not forget to share your model on huggingface.co/models =)

{'train_runtime': 1076.5441, 'train_samples_per_second': 6.957, 'train_steps_per_second': 0.102, 'train_loss': 0.6166326316920194, 'epoch': 9.36}

**Comment by iMountTai** [comment]
试试BF16

**Comment by litsh** [comment]
没用啊




Thaison
***@***.***







                       
Original Email
                       
                     

Sender:"Xin Yao"< ***@***.*** &gt;;

Sent Time:2023/10/26 21:01

To:"ymcui/Chinese-LLaMA-Alpaca-2"< ***@***.*** &gt;;

Cc recipient:"litsh"< ***@***.*** &gt;;"Author"< ***@***.*** &gt;;

Subject:Re: [ymcui/Chinese-LLaMA-Alpaca-2] 整个训练过程中&#39;eval_loss&#39;都是nan，但是training_loss正常，一直在下降 (Issue #370)




 
试试BF16

**Comment by litsh** [comment]
我看到好多人都有这个问题，不知道有没有人解决了

**Comment by tkone2018** [comment]
@litsh 我也出现这个问题了

**Comment by litsh** [comment]
> @litsh 我也出现这个问题了

你解决了吗

**Comment by yusufcakmakk** [comment]
I also have this problem. Is there any solution?

**Comment by qdchenxiaoyan** [comment]
我是eval_loss随着训练一直增大，不知道是什么原因，从1.9训练10轮到了2.7

**Comment by 5663015** [comment]
看到好几个issue都是这个问题，我也遇到了这个问题
